From 9bee62e2eae153e9678319d52b35af63d417575d Mon Sep 17 00:00:00 2001
From: "C, Udhayakumar" <udhayakumar.c@intel.com>
Date: Mon, 24 May 2021 17:12:42 +0530
Subject: [PATCH 038/109] Thermal zone unregister fix in tsens host driver.

Thermal zone pointer is made as NULL and passed for unregister.
Hence thermal zone is not removed properly on reset's.

Signed-off-by: C, Udhayakumar <udhayakumar.c@intel.com>
---
 drivers/misc/intel_tsens/intel_tsens_host.c | 2 +-
 drivers/misc/xlink-smbus/xlink-smbus.c      | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/misc/intel_tsens/intel_tsens_host.c b/drivers/misc/intel_tsens/intel_tsens_host.c
index 8d3fbf741524..77868e2cddd8 100644
--- a/drivers/misc/intel_tsens/intel_tsens_host.c
+++ b/drivers/misc/intel_tsens/intel_tsens_host.c
@@ -293,7 +293,7 @@ static void intel_tsens_remove_tz(struct intel_hddl_clients *d)
 		}
 		if (tsens->tz_xlk) {
 			mutex_lock(&tsens->sync_xlk_unregister);
-			tsens->tz_xlk = NULL;
+			tsens->i2c_xlk = NULL;
 			thermal_zone_device_unregister(tsens->tz_xlk);
 			mutex_unlock(&tsens->sync_xlk_unregister);
 		}
diff --git a/drivers/misc/xlink-smbus/xlink-smbus.c b/drivers/misc/xlink-smbus/xlink-smbus.c
index fc652e6c96bb..4ecfe42ad765 100644
--- a/drivers/misc/xlink-smbus/xlink-smbus.c
+++ b/drivers/misc/xlink-smbus/xlink-smbus.c
@@ -158,6 +158,8 @@ static s32 xlink_smbus_xfer(struct i2c_adapter *adap, u16 addr,
 	if (!adap)
 		return -ENODEV;
 	adapt_data = i2c_get_adapdata(adap);
+	if (!adapt_data)
+		return -ENODEV;
 	dev = &adapt_data->adap->dev;
 
 	if (!data)
-- 
2.25.1

